### PR TITLE
add pokedex component and seed data

### PR DESCRIPTION
**Pull request by nathanSamosa**

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,4 +1,4 @@
 DATABASE_URL="YOUR_DATABASE_URL?schema=prisma"
 SHADOW_DATABASE_URL="YOUR_SHADOW_DATABASE_URL?schema=shadow"
-API_URL="http://localhost:4000"
+REACT_APP_API_URL="http://localhost:4000"
```
Comment (Can-Tina) [inline .env.example:3]: Nicely figured out, I know that was tricky

File: prisma/schema.prisma [hunk 2]
```diff
@@ -38,7 +38,7 @@ model User {
 model Profile {
   id             Int              @id @default(autoincrement())
   user           User             @relation(fields: [userId], references: [id])
-  userId         Int              @unique
+  userId         Int
```
Comment (Can-Tina) [inline prisma/schema.prisma]: since the profile is one to one, why wouldn't the userId be unique?

File: prisma/schema.prisma [hunk 3]
```diff
@@ -70,20 +70,11 @@ model Rating {
...
-}
-
 model Post {
   id         Int       @id @default(autoincrement())
   profile    Profile   @relation(fields: [profileId], references: [id])
-  profileId  Int       @unique
-  category   Category  @relation(fields: [categoryId], references: [id])
-  categoryId Int       @unique
+  profileId  Int       
+  category   String
```
Comment (Can-Tina) [inline prisma/schema.prisma:77]: This does make more sense than a random category table. Nice

File: prisma/schema.prisma [hunk 4]
```diff
@@ -94,9 +85,9 @@ model Post {
 model Comment {
   id        Int      @id @default(autoincrement())
   post      Post     @relation(fields: [postId], references: [id])
-  postId    Int      @unique
+  postId    Int      
   profile   Profile  @relation(fields: [profileId], references: [id])
-  profileId Int      @unique
+  profileId Int
```
Comment (Can-Tina) [inline prisma/schema.prisma:90]: removing unique here does make sense

File: src/client/components/Pokedex.jsx [hunk 5]
```diff
@@ -1,7 +1,104 @@
+import { useState, useEffect } from "react";
+
+import "../styling/pokedex.css"
+import ratingStar from "../assets/star.png"
+import ratingStarHollow from "../assets/Star_Outline_2.png"
+
+const URL = process.env.REACT_APP_API_URL;
```
Comment (Can-Tina) [inline src/client/components/Pokedex.jsx:7]: No more hardcoding! Nice :D

File: src/client/components/Pokedex.jsx [hunk 6]
```diff
@@ -1,7 +1,104 @@
...
+const pokemonEndpoint = "/pokemon";
```
Comment (Can-Tina) [inline src/client/components/Pokedex.jsx:8]: No "Magic Strings". Nice :)

File: src/server/controllers/user.js [hunk 7]
```diff
@@ -15,7 +15,10 @@ const postRegisterDetails = async (req, res) => {
             data: {
                 email: email,
                 username: username,
-                password: hashedPassword
+                password: hashedPassword,
+                profile: {
+                    crate: {}
```
Comment (Can-Tina) [inline src/server/controllers/user.js]: Is this not supposed to be `create` instead of `crate`?

File: prisma/schema.prisma (hunk 2 above)
Comment (nathanSamosa) [reply prisma/schema.prisma]: I removed it when troubleshooting, yeah it should still be unique

File: src/server/controllers/user.js (hunk 7 above)
Comment (nathanSamosa) [reply src/server/controllers/user.js]: oops! sorted

File: src/client/components/Pokedex.jsx (hunk 5 above)
Comment (TheunisK) [reply src/client/components/Pokedex.jsx:7]: You tell him Lenni.

File: src/server/controllers/user.js (hunk 7 above)
Comment (TheunisK) [reply src/server/controllers/user.js]: Good spot

File: .env.example (hunk 1 above)
Comment (TheunisK) [reply .env.example:3]: Yes nice! Let me know when you figure out "process("dotenv").config()"? I think thats the way to get it in express? If I understadn correctly.

File: prisma/schema.prisma (hunk 4 above)
Comment (TheunisK) [reply prisma/schema.prisma:90]: How will the profile id not be unique? I dont get how two profiles can have the same id?

File: .env.example (hunk 1 above)
Comment (glowkeeper) [reply .env.example:3]:
```
require('dotenv').config()
console.log("My URL", process.env.REACT_APP_API_URL)

```

File: prisma/schema.prisma (hunk 2 above)
Comment (glowkeeper) [reply prisma/schema.prisma]: defo should be unique - enforce the constraint, don't just assume it ;)

File: prisma/schema.prisma (hunk 4 above)
Comment (glowkeeper) [reply prisma/schema.prisma:90]: see comment above - I reckon this should be unique...

File: src/client/components/Pokedex.jsx (hunk 6 above)
Comment (glowkeeper) [reply src/client/components/Pokedex.jsx:8]: but isn't this a magic string? `const pokemonEndpoint = "/pokemon";`

**Comment by glowkeeper** [comment]
Er, tempted to approve, except I reckon `profileId` should be unique. @Can-Tina - why did you remove that constraint?

File: prisma/schema.prisma (hunk 4 above)
Comment (nathanSamosa) [reply prisma/schema.prisma:90]: this was on the comment model; if it were unique, users could only have 1 comment!

Comment (Can-Tina) [reply prisma/schema.prisma:90]: Yeah, I've had problems with unique constraints with the ratings model. It wouldn't work if it was unique

File: src/client/components/Pokedex.jsx (hunk 6 above)
Comment (Can-Tina) [reply src/client/components/Pokedex.jsx:8]: but it's in a constant right?

**Comment by Can-Tina** [comment]
It wouldn't work if it was, as Nathan explained each user would only be able to make one comment

**Comment by glowkeeper** [comment]
Ah - okay - so this is a foreign key in the comment table then, I take it. In that case, approved :)

File: src/client/components/Pokedex.jsx (hunk 6 above)
Comment (glowkeeper) [reply src/client/components/Pokedex.jsx:8]: I suppose - and indeed, that's better - but I'd prefer all my string constants in some string constants file that I then import where necessary. That way, I know where to find them all.

**Pull request merged**